spring-security-cas-extension/src/main/java/com/kakawait/spring/security/cas/client/AuthenticatedPrincipal.java

```diff
@@ -0,0 +1,11 @@
+package com.kakawait.spring.security.cas.client;
+
+import java.security.Principal;
+
+/**
+ * @author Jonathan Coueraud
+ */
+public interface AuthenticatedPrincipal {

```

Comment:
I don't really fan of that `name` because it look like that interface hold the `AuthenticatedPrincipal` where is just an interface that abstract how to retrieve the `Principal`.

For now I don't have other name but I would think about but the idea will be more around `PrincipalExtractor` (but I don't like that name too)